src/parallel/boundaries/BoundaryUtils.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * BoundaryUtils.h
+ *
+ *  Created on: 24 March 2023
+ *      Author: amartyads
+ */
+
+#pragma once
+
+#include <array>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "molecules/Molecule.h"
+
+/**
+ * Includes enums and helper functions for processing boundary conditions.
+ *
+ * Does not actually work on any molecule containers, hence can be used
+ * in a general sense for other calculations.
+ */
+namespace BoundaryUtils {
+
+/**
+ * enum storing the types of boundaries currently supported.
+ *
+ * The currently supported boundary types are
+ * PERIODIC_OR_LOCAL - periodic boundaries or local boundaries: both of which
+ * use the default (communicate particle transfers with neighbours) behaviour
+ * OUTFLOW - molecules exiting the boundary are deleted
+ * REFLECTING - molecules exiting the boundary have their velocities
+ * reversed in the direction they are leaving
+ * ERROR - kept for sanity checks
+ *
+ * This can be extended if needed.
+ */
+enum class BoundaryType { PERIODIC_OR_LOCAL, OUTFLOW, REFLECTING, ERROR };
+
+/**
+ * enum storing the axes and direction.
+ *
+ * The dimensions are POSX, NEGX, POSY, NEGY, POSZ and NEGZ.
+ *
+ * This is hardcoded for 3D, and ERROR is included for sanity checks.
+ */
+enum class DimensionType { POSX, NEGX, POSY, NEGY, POSZ, NEGZ, ERROR };
```

Comment:
I am aware I could have used the enums in smarter ways, mapped the int values to make the conversions oneliners. I specifically used the enums and made everything super verbose so that in the future
- my intentions are laid out super clearly, so that even if the code looks like "baby's first mobile app enums" there's no confusion
- coders will be forced to use the converters to mess with boundaries, and not do smart things (like use `0,0,0` for `globalRegionBegin` instead of calling the correct functions)
I was using that guiding philosophy for most of the code, hence the verbosity. So I don't want to change the enums, but most of the other ideas are strict upgrades, so I'll use them.